apps/src/componentLibrary/dropdown/iconDropdown/IconDropdown.tsx

```diff
@@ -0,0 +1,142 @@
+import React, {useCallback, memo} from 'react';
+import classNames from 'classnames';
+
+import {
+  ComponentSizeXSToL,
+  DropdownColor,
+} from '@cdo/apps/componentLibrary/common/types';
+import moduleStyles from '@cdo/apps/componentLibrary/dropdown/customDropdown.module.scss';
+
+import CustomDropdown from '@cdo/apps/componentLibrary/dropdown/_CustomDropdown';
+
+import FontAwesomeV6Icon, {
+  FontAwesomeV6IconProps,
+} from '@cdo/apps/componentLibrary/fontAwesomeV6Icon';
+
+import {dropdownColors} from '@cdo/apps/componentLibrary/common/constants';
+import {
+  DropdownProviderWrapper,
+  useDropdownContext,
+} from '@cdo/apps/componentLibrary/common/contexts/DropdownContext';
+
+export interface IconDropdownOption {
+  value: string;
+  label: string;
+  isOptionDisabled?: boolean;
+  icon: FontAwesomeV6IconProps;
+}
+
+export interface IconDropdownProps {
+  /** IconDropdown name */
+  name: string;
+  /** IconDropdown color */
+  color?: DropdownColor;
+  /** IconDropdown size */
+  size: ComponentSizeXSToL;
+  /** IconDropdown disabled state */
+  disabled?: boolean;
+  /** IconDropdown label */
+  labelText: string;
+  /** IconDropdown options */
+  options: IconDropdownOption[];
+  /** IconDropdown checked options */
+  selectedOption: IconDropdownOption;
+  /** IconDropdown onChange handler */
+  onChange: (option: IconDropdownOption) => void;
+}
+
+/**
+ * ### Production-ready Checklist:
+ * * (✔) implementation of component approved by design team;
+ * * (✔) has storybook, covered with stories and documentation;
+ * * (✔) has tests: test every prop, every state and every interaction that's js related;
+ * * (see apps/test/unit/componentLibrary/DropdownMenuTest.jsx)
+ * * (?) passes accessibility checks;
+ *
+ * ###  Status: ```Ready for dev```
+ *
+ * Design System: Icon Dropdown Component.
+ * Used to render dropdowns with a list of options with icons.
+ */
+const IconDropdown: React.FunctionComponent<IconDropdownProps> = ({
+  name,
+  labelText,
+  options,
+  selectedOption = {},
+  onChange,
+  disabled = false,
+  color = dropdownColors.black,
+  size = 'm',
+}) => {
+  const {setActiveDropdownName} = useDropdownContext();
+  const onOptionClick = useCallback(
+    (option: IconDropdownOption) => {
+      if (!disabled && !option.isOptionDisabled) {
+        onChange(option);
+        setActiveDropdownName('');
+      }
+    },
+    [disabled, onChange, setActiveDropdownName]
+  );
+
+  return (
+    <CustomDropdown
+      name={name}
+      labelText={labelText}
+      disabled={disabled}
+      color={color}
+      // isSomeValueSelected={true}
+      icon={selectedOption?.icon}
+      size={size}
+    >
+      <form className={moduleStyles.dropdownMenuContainer}>
+        <ul>
+          {options.map(option => {
```

Comment:
A lot of logic in this function, might be nice to move into its own helper?

Reply:
Don't have super strong opinion on this one, but I believe that it's easier to read when all the IconDropdown is gathered in one place. Giving that IconDropdown is less than 150 lines of code, and moving this logic into separate function will require adding more props into it and making it harder to understand the logic. Also, this is mostly the logic representing IconDropdown itself, meanwhile other code is rather 'template' dropdown code

Leaning towards keeping as is.